gfx_scene.cpp

```diff
@@ -2630,7 +2698,11 @@ class GfxSceneInternal
                 mipDepth = GFX_MAX(1U, mipDepth / 2);
             }
         }
-        image_ref->flags = (image_ref->channel_count != 4 ? 0 : kGfxImageFlag_HasAlphaChannel);
+        image_ref->flags = (image_ref->channel_count != 4
+            || (image_ref->format == DXGI_FORMAT_BC7_TYPELESS
+            || image_ref->format == DXGI_FORMAT_BC7_UNORM
+            || image_ref->format == DXGI_FORMAT_BC7_UNORM_SRGB) //BC7 may or may not have alpha
+            ? 0 : kGfxImageFlag_HasAlphaChannel);
```

Comment:
We now only set the has alpha channel flag if we are sure one exists. BC7 can either have one or not depending on the compression mode so there is no way of knowing. Instead of always flagging an alpha we default to not flagging it in the case of BC7, this may still be wrong but if using BC7 albedo one should use the new alpha_mode flag to indicate of alpha testing should be performed